File: 739-daily-temperatures/739-daily-temperatures.py
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        n = len(temperatures)
        res = n * [0]
        stack = []
        for i in range(n-1, -1, -1):
            while (stack and (temperatures[i] >= temperatures[stack[-1]])):
                stack.pop()
            if not(stack):
                res[i] = 0
            else:
                res[i] = stack[-1] - i
            stack.append(i)
        return res

File: 739-daily-temperatures/739-daily-temperatures.rs
impl Solution {
    pub fn daily_temperatures(temperatures: Vec<i32>) -> Vec<i32> {
        let n: i32 = temperatures.len() as i32;
        let mut res: Vec<i32> = Vec::new();
        for i in 0..n {
            res.push(0);
        }
        let mut stack: Vec<i32> = Vec::new();
        for i in (0..n).rev() {
            while (!(stack.is_empty())) {
                if (temperatures[i as usize] >= temperatures[stack[(stack.len() - 1) as usize] as usize]) {
                    stack.pop();
                }
                else {
                    break;
                }
            }
            if (stack.is_empty()) {
                res[i as usize] = 0;
            }
            else {
                res[i as usize] = stack[(stack.len() - 1) as usize] - i;
            }
            stack.push(i);
        }
        return res;
    }
}
